### PR TITLE
Fix percentage distance methods

### PR DESCRIPTION
**Pull request by JackB-Ansys**
Deprecate get_coordinate_from_percentage_distance methods because these actually use fractional distances. Replace this functionality by adding optional arguments for get_coordinate_from_distance, where user can specify a percentage or fraction instead of specifying an absolute distance.

### CODE REVIEW

**Comment by jgsdavies** [comment]
> ## [Codecov](https://app.codecov.io/gh/ansys/pymotorcad/pull/339?dropdown=coverage&src=pr&el=h1&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=ansys) Report
> Attention: Patch coverage is `74.19355%` with `8 lines` in your changes missing coverage. Please review.
> 
> > Project coverage is 84.10%. Comparing base [(`a3afd47`)](https://app.codecov.io/gh/ansys/pymotorcad/commit/a3afd4781ca390e5d48775cbffda67050424ea56?dropdown=coverage&el=desc&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=ansys) to head [(`0f9f663`)](https://app.codecov.io/gh/ansys/pymotorcad/commit/0f9f6632640c175821e3358c41d1b5e5145a047c?dropdown=coverage&el=desc&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=ansys).
> > Report is 2 commits behind head on main.
> 
> Additional details and impacted files

@JackB-Ansys You will need to improve the existing tests to cover all the new lines of code

File: src/ansys/motorcad/core/geometry.py [hunk 1]
```diff
@@ -918,51 +918,63 @@ def mirror(self, mirror_line):
         else:
             raise Exception("Line can only be mirrored about Line()")
 
-    def get_coordinate_from_percentage_distance(self, ref_coordinate, percentage):
-        """Get the coordinate at the percentage distance along the line from the reference.
+    def get_coordinate_from_percentage_distance(self, ref_coordinate, fraction):
+        """Get the coordinate at a fractional distance along the line from the reference coord.
+
+        .. note::
```
Comment (jgsdavies) [inline src/ansys/motorcad/core/geometry.py:924]: Please also add this info to a deprecation warning 
https://docs.python.org/3/library/exceptions.html#DeprecationWarning

File: src/ansys/motorcad/core/geometry.py [hunk 2]
```diff
@@ -918,51 +918,63 @@ def mirror(self, mirror_line):
...
-
-        t = (self.length * percentage) / self.length
-        x = ((1 - t) * coordinate_1.x) + (t * coordinate_2.x)
-        y = ((1 - t) * coordinate_1.y) + (t * coordinate_2.y)
+        return self.get_coordinate_from_distance(ref_coordinate, fraction=fraction)
 
-        return Coordinate(x, y)
-
-    def get_coordinate_from_distance(self, ref_coordinate, distance):
+    def get_coordinate_from_distance(
```
Comment (jgsdavies) [inline src/ansys/motorcad/core/geometry.py:943]: I like this change. @matthewAnsys thoughts?

File: tests/test_geometry.py [hunk 3]
```diff
@@ -531,14 +531,31 @@ def test_line_length():
     assert line.length == sqrt(2)
 
 
-def test_arc_get_coordinate_from_percentage_distance():
+def test_arc_get_coordinate_from_fractional_distance():
```
Comment (jgsdavies) [inline tests/test_geometry.py:534]: Can we make sure this also works for arcs with negative radius

**Review by jgsdavies — CHANGES_REQUESTED** [review]

**Comment by jgsdavies** [comment]
Needs merge from main

File: tests/test_geometry.py (hunk 3 above)
Comment (JackB-Ansys) [reply tests/test_geometry.py:603]: Tests added for this

File: src/ansys/motorcad/core/geometry.py [hunk 4]
```diff
@@ -943,51 +944,74 @@ def mirror(self, mirror_line):
...
+
+        percentage : float, optional
+            Percentage distance along Line.
+
         Returns
         -------
         Coordinate
             Coordinate at distance along Line.
         """
+        if not distance and not fraction and not percentage:
```
[the remark this answers is not in the document]
Comment (JackB-Ansys) [reply src/ansys/motorcad/core/geometry.py:999]: addressed in latest commit

File: src/ansys/motorcad/core/geometry.py [hunk 5]
```diff
@@ -998,51 +999,74 @@ def mirror(self, mirror_line):
...
-        percentage : float
-            Percentage distance along Line.
+        fraction : float
+            Fractional distance along Arc.
 
         Returns
         -------
         Coordinate
-            Coordinate at percentage distance along Line.
+            Coordinate at fractional distance along Arc.
```
Comment (james-packer) [inline src/ansys/motorcad/core/geometry.py:1020]:
```suggestion
            Coordinate at fractional distance along Line.
```

File: src/ansys/motorcad/core/geometry.py [hunk 6]
```diff
@@ -998,51 +999,74 @@ def mirror(self, mirror_line):
...
 
         Parameters
         ----------
         ref_coordinate : Coordinate
             Entity reference coordinate.
 
-        percentage : float
-            Percentage distance along Line.
+        fraction : float
+            Fractional distance along Arc.
```
Comment (james-packer) [inline src/ansys/motorcad/core/geometry.py:1015]:
```suggestion
            Fractional distance along Line.
```

File: src/ansys/motorcad/core/geometry.py [hunk 7]
```diff
@@ -1162,42 +1186,74 @@ def midpoint(self):
...
         ----------
         ref_coordinate : Coordinate
            Entity reference coordinate.
 
-        distance : float
+        distance : float, optional
             Distance along arc.
 
+        fraction : float, optional
+            Fractional distance along Line.
```
Comment (james-packer) [inline src/ansys/motorcad/core/geometry.py:1231]:
```suggestion
            Fractional distance along Arc.
```

File: src/ansys/motorcad/core/geometry.py [hunk 8]
```diff
@@ -1162,42 +1186,74 @@ def midpoint(self):
...
+
+        percentage : float, optional
+            Percentage distance along Line.
```
Comment (james-packer) [inline src/ansys/motorcad/core/geometry.py:1234]:
```suggestion
            Percentage distance along Arc.
```

File: src/ansys/motorcad/core/geometry.py [hunk 9]
```diff
@@ -998,51 +999,74 @@ def mirror(self, mirror_line):
         else:
             raise Exception("Line can only be mirrored about Line()")
 
-    def get_coordinate_from_percentage_distance(self, ref_coordinate, percentage):
-        """Get the coordinate at the percentage distance along the line from the reference.
+    def get_coordinate_from_percentage_distance(self, ref_coordinate, fraction):
+        """Get the coordinate at a fractional distance along the line from the reference coord.
+
+        .. note::
+           This method is deprecated. Use the :func:`Arc.get_coordinate_from_distance`
```
Comment (james-packer) [inline src/ansys/motorcad/core/geometry.py:1006]:
```suggestion
           This method is deprecated. Use the :func:`Line.get_coordinate_from_distance`
```

**Pull request merged**